### PR TITLE
tosproc: completely remove compiler execution

### PR DESCRIPTION
**Pull request by alaviss**
Previously this test would depend on the compiler being available to execute to recompile itself.

Not only is this hugely inefficient, but it also produce a dependency on the compiler being in `PATH`.

Instead this pull request:

- Fold external runners into the same test binary. This removes the need of recompilation altogether.
- Replaced `nim r` in tests with much simpler "consume all" runner that allow testing the same thing. This prevents compiler bugs from tripping this test up.

Done as part of #89 to remove `PATH` dependency from test suite.

### CODE REVIEW

File: tests/stdlib/tosproc.nim [hunk 1]
```diff
@@ -4,60 +4,60 @@ joinable: false
...
+    if true: quit(0)
+  of "exitnow_139":
+    if true: exitnow(139)
+  of "c_exit2_139":
+    if true: c_exit2(139)
+  of "quit_139":
+    # `exitStatusLikeShell` doesn't distinguish between a process that
+    # exit(139) and a process that gets killed with `SIGSEGV` because
+    # 139 = 11 + 128 = SIGSEGV + 128.
+    # However, as #10249 shows, this leads to bad debugging experience
```
Comment (saem) [inline tests/stdlib/tosproc.nim]: Should we try to add repo qualifiers for the bug references?

File: tests/stdlib/tosproc.nim [hunk 2]
```diff
@@ -4,60 +4,60 @@ joinable: false
 
 #[
 joinable: false
-because it'd need cleanup up stdout
+because it has to execute itself
```
Comment (saem) [inline tests/stdlib/tosproc.nim:7]: I think this would be better if the runner supported an array argument sets to pass a test so it could call it repeatedly instead. Although in this case we'd need more for the whole input output bit? Also scope creep.

**Review by saem — APPROVED** [review]
Minor comments, I think I get it. It's better than before, that's for sure.

File: tests/stdlib/tosproc.nim (hunk 2 above)
Comment (alaviss) [reply tests/stdlib/tosproc.nim:7]: This test tests `osproc`, and it executes itself with `osproc.*` to run its subprogram to verify behaviors. It has nothing to do with having to run with different arguments :P

Comment (saem) [reply tests/stdlib/tosproc.nim:7]: Wait, the sub-programs are itself, right? Or did I read that wrong?

Anyhow, I was considering it for each of the permutation that it would run, would still need to have other processes and all that. Just that the reporting could be a little bit more fine grained.

Comment (alaviss) [reply tests/stdlib/tosproc.nim:7]: Ah yea, we can certainly add a runner <-> test IPC in the future.
